sematic/ui/packages/common/src/typeViz/views/hugging_face_stored_model.tsx

```diff
@@ -0,0 +1,27 @@
+import { ValueComponentProps, ViewComponentProps} from "src/typeViz/common";
+
+export function HuggingFaceStoredModelShortView(props: ValueComponentProps) {
+    return HuggingFaceStoredModelView(props, true);
+}
+
+export function HuggingFaceStoredModelFullView(props: ViewComponentProps) {
+    return HuggingFaceStoredModelView(props, false);
+}
+
+function HuggingFaceStoredModelView(props: ValueComponentProps | ViewComponentProps, short: boolean) {
+    const { valueSummary } = props;
+    const { values } = valueSummary;
+    const modelType = values.peft_model_type ? values.peft_model_type : values.model_type;
```

Comment:
If we have this as a functional component, then we can do 

```
const contents = useMemo(() => {
        const modelType = values.peft_model_type ? values.peft_model_type : values.model_type;
        const modelTypePieces = modelType.split(".");
        const modelTypeShortName = modelTypePieces[modelTypePieces.length - 1];

        return (
            short ?
                "model" :
                (<span><strong>{modelTypeShortName}</strong>: {values.path}</span>)
        )
    }, [values, short]);
```
to avoid performance loss due to re-rendering.

Reply:
done